bigframes/streaming/__init__.py

```diff
@@ -13,17 +13,254 @@
+            self.sql,
+            instance=instance,
+            table=table,
+            service_account_email=service_account_email,
+            session=self._session,
+            app_profile=app_profile,
+            truncate=truncate,
+            overwrite=overwrite,
+            auto_create_column_families=auto_create_column_families,
+            bigtable_options=bigtable_options,
+            job_id=job_id,
+            job_id_prefix=job_id_prefix,
+        )
+
+    def to_pubsub(
+        self,
+        *,
+        topic: str,
+        service_account_email: str,
+        job_id: Optional[str] = None,
+        job_id_prefix: Optional[str] = None,
+    ) -> bigquery.QueryJob:
+        """
+        Export the StreamingDataFrame as a continue job and returns a
+        QueryJob object for some management functionality.
+
+        This method requires an existing pubsub topic. For instructions
+        on creating a pubsub topic, see
+        https://cloud.google.com/pubsub/docs/samples/pubsub-quickstart-create-topic?hl=en
+
+        Note that a service account is a requirement for continuous queries
+        exporting to pubsub.
+
+        Args:
+            topic (str):
+                The name of the pubsub topic to export to.
+                For example: "taxi-rides"
+            service_account_email (str):
+                Full name of the service account to run the continuous query.
+                Example: accountname@projectname.gserviceaccounts.com
+            job_id (str, default None):
+                If specified, replace the default job id for the query,
+                see job_id parameter of
+                https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.client.Client#google_cloud_bigquery_client_Client_query
+            job_id_prefix (str, default None):
+                If specified, a job id prefix for the query, see
+                job_id_prefix parameter of
+                https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.client.Client#google_cloud_bigquery_client_Client_query
+
+        Returns:
+            google.cloud.bigquery.QueryJob:
+                See https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.job.QueryJob
+                The ongoing query job can be managed using this object.
+                For example, the job can be cancelled or its error status
+                can be examined.
+        """
+        return _to_pubsub(
+            self.sql,
+            topic=topic,
+            service_account_email=service_account_email,
+            session=self._session,
+            job_id=job_id,
+            job_id_prefix=job_id_prefix,
+        )
+
+
+class StreamingDataFrame(StreamingBase):
+    __doc__ = _curate_df_doc(dataframe.DataFrame.__doc__)
+
+    # Private constructor
+    _create_key = object()
+
+    def __init__(self, df: dataframe.DataFrame, *, create_key=0):
```

Comment:
I would really not wrap the main dataframe.DataFrame object. This reduces flexibility to modify this object - as it could break streaming behavior by accident.

Reply:
I think this is the best way at least for current moment. That we want to get some SDF APIs working. If any code is sharable I would like to try to share instead of duplicate. It would be better to find something breaking and fix at first place than let the implementations diverge and hard to find where they are different and cause separate issues.

Now with disabling snapshot and caching, we can see some operations can just work. I think it is decent for it now. Later on, Tardis will increase SQL coverage. And we will want to add more APIs. We may have a better idea of how diverge they need to be.